default/python/universe_generation/specials.py

```diff
@@ -1,76 +1,188 @@
 import random
-import freeorion as fo
+from collections import defaultdict
 import statistics
+
+import freeorion as fo
 import universe_tables
+from timers import Timer
+from galaxy_topology import get_systems_within_jumps
+
+# REPEAT_RATE along with calc_num_placed determines if there are multiple specials in a single
+# location.  There can only be at most 4 specials in a single location.
+# The probabilites break down as follows:
+# Count  Probability
+# one    (1 - REPEAT_RATE[0])
+# two    REPEAT_RATE[0] * (1 - REPEAT_RATE[1])
+# three  REPEAT_RATE[0] * REPEAT_RATE[1] *(1 - REPEAT_RATE[2])
+# four   REPEAT_RATE[0] * REPEAT_RATE[1] * REPEAT_RATE[2]
+REPEAT_RATE = {1: 0.08, 2: 0.05, 3: 0.01, 4: 0.00}
+
+
+def calc_num_placed(objs):
+    """Return a list of number placed at each obj"""
+    return [1 if random.random() > REPEAT_RATE[1] else
+            2 if random.random() > REPEAT_RATE[2] else
+            3 if random.random() > REPEAT_RATE[3] else 4
+            for __ in objs]
+
+
+def place_special(specials, obj, specials_timer, per_special_timer):
+    """ Place at most a single special.
```

Comment:
I can't see when you would place 0 specials.

After your normalization, the sum of thresholds is exactly 1. random.random() returns a number [0, 1). 
If no special was previously performed, the last if check performed is 
```chance - (sum(thresholds)-threshold[end])  = chance - (1-threshold[end]) > threshold[end] <--> chance > 1```
which is false in any case.
Therefore, you will *always* place a special with the given code (unless my math failed me somewhere).